src/completion-provider/script-completion-provider.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  CompletionItem,
+  TextDocumentPositionParams
+} from 'vscode-languageserver';
+
+import Server from '../server';
+import ASTPath from '../glimmer-utils';
+import { toPosition } from '../estree-utils';
+import { filter } from 'fuzzaldrin';
+import { parseScriptFile as parse } from 'ember-meta-explorer';
+const { uniqBy } = require('lodash');
+const memoize = require('memoizee');
+import { getExtension } from '../utils/file-extension';
+import { log } from '../utils/logger';
+import {
+  isStoreModelLookup,
+  isRouteLookup,
+  isModelReference,
+  isNamedServiceInjection,
+  isTransformReference,
+  isComputedPropertyArgument
+} from '../utils/ast-helpers';
+import {
+  listRoutes,
+  listModels,
+  listServices,
+  mGetProjectAddonsInfo,
+  listTransforms
+} from '../utils/layout-helpers';
+
+const mListRoutes = memoize(listRoutes, { length: 1, maxAge: 60000 });
```

Comment:
As we merge this, I'm going to go ahead and remove this memoization. 10 minutes is *far* too long, and the need for it implies that we have some other architectural issues we need to fix. I'm going to drop these, then run the profiler against the LS to see why it breaks down. As a rule, that's much preferable to hard-coding in limits like this!